### PR TITLE
Update mcr.cpp

### PR DESCRIPTION
**Pull request by aiwosiwo**
- Variables ‘player1’ and ‘player2’ are deleted
- Logical error in determining a tie is corrected 
- IsWin function is simplified
- Add repeated judgment and range judgment on the input
- Correct input hints
- Simplify output of final grid

### CODE REVIEW

File: mcr.cpp [hunk 1]
```diff
@@ -2,48 +2,65 @@
 using namespace std;
 
 bool isWin(char game[3][3]){
-bool win = false;
```
Comment (aiwosiwo) [inline mcr.cpp:5]: 这里改了就不会有冗余判断，节省时间和算力

**Review by aiwosiwo — COMMENTED** [review]
isWin函数里删除了多余变量win

File: mcr.cpp [hunk 2]
```diff
@@ -2,48 +2,65 @@
 using namespace std;
 
 bool isWin(char game[3][3]){
-bool win = false; 
-    if (game[0][0] == game[0][1] && game[0][1] == game[0][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true; 
-    if (game[1][0] == game[1][1] && game[1][1] == game[1][2] && (game[1][0] == 'X' || game[1][0] == 'O')) win = true; 
-    if (game[2][0] == game[2][1] && game[2][1] == game[2][2] && (game[2][0] == 'X' || game[2][0] == 'O')) win = true; 
+    if (game[0][0] == game[0][1] && game[0][1] == game[0][2] && (game[0][0] == 'X' || game[0][0] == 'O')) return true;
```
Comment (aiwosiwo) [inline mcr.cpp:5]: 这里return true换行写吧，写到一行不易查看

File: mcr.cpp [hunk 3]
```diff
@@ -2,48 +2,65 @@
...
+    if (game[0][0] == game[1][0] && game[1][0] == game[2][0] && (game[0][0] == 'X' || game[0][0] == 'O')) return true;
+    if (game[0][1] == game[1][1] && game[1][1] == game[2][1] && (game[0][1] == 'X' || game[0][1] == 'O')) return true; 
+    if (game[0][2] == game[1][2] && game[1][2] == game[2][2] && (game[0][2] == 'X' || game[0][2] == 'O')) return true; 
     // diagonal 
-    if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O')) win = true; 
-    if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O')) win = true; 
-    return win;
-} 
+    if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] == 'X' || game[0][0] == 'O')) return true; 
+    if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] == 'X' || game[0][2] == 'O')) return true;
```
Comment (MioAyanokouji) [inline mcr.cpp:14]: 可以使用isFill(game, i, j)去替代(game[0][0] == 'X' || game[0][0] == 'O')的判断

File: mcr.cpp [hunk 4]
```diff
@@ -2,48 +2,65 @@
...
+        while (i < 0 || i > 2 || j < 0 || j > 2){ //range judgment
+            cout << "Out of range, input again: ";
+            cin >> i >> j;
+        }
+        while (isFill(game, i, j)){ //repeat judgment
+            cout << "Already filled, input again: ";
+            cin >> i >> j;
+        }
+        if (turn == false)
             game[i][j] = 'X';
```
Comment (MioAyanokouji) [inline mcr.cpp:49]: game[i][j] = player1;

File: mcr.cpp (hunk 3 above)
Comment (aiwosiwo) [reply mcr.cpp:14]: 可，我试试

File: mcr.cpp (hunk 2 above)
Comment (aiwosiwo) [reply mcr.cpp:5]: 还是不改了

**Pull request merged**